test/test_onnx.py

```
@@ -3,63 +3,376 @@
+                segment_codes = onnx_output[0, :].tolist()
+            elif onnx_output.ndim == 3:
+                segment_codes = onnx_output[0, 0, :].tolist()
+            else:
+                segment_codes = onnx_output.tolist()
+
+            segment_results.append(segment_codes)
+
+        except Exception as e:
+            print(f"  ONNX error on segment {i+1}: {str(e)[:100]}...")
+            raise Exception(
+                f"ONNX inference failed on segment {i+1}: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)